AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -651,13 +665,18 @@ - (NSIndexPath *)indexPathForNode:(ASCellNode *)cellNode;
 - (NSArray *)nodesAtIndexPaths:(NSArray *)indexPaths
 {
   ASDisplayNodeAssertMainThread();
-  return ASFindElementsInMultidimensionalArrayAtIndexPaths(_completedNodes, [indexPaths sortedArrayUsingSelector:@selector(compare:)]);
+  return ASFindElementsInMultidimensionalArrayAtIndexPaths([self externalCompletedNodes], [indexPaths sortedArrayUsingSelector:@selector(compare:)]);
 }
 
 - (NSArray *)completedNodes
 {
   ASDisplayNodeAssertMainThread();
-  return _completedNodes;
+  return [self externalCompletedNodes];
+}
+
+- (NSMutableArray *)externalCompletedNodes
+{
+  return _externalCompletedNodes != nil ? _externalCompletedNodes : _completedNodes;
```

Comment:
Move the ASDisplayNodeAssertMainThread() to this method.

I'm wondering if the existing accessor -completedNodes makes sense.  I know it is API exposed on ASDataController, I think used by ASRangeController.  Right now the names between these don't quite sit right with me, particularly because the external one is mutable, and also is not /actually/ externally exposed at all (at least the method name).

Could we just delete one of these methods, and rename externalCompletedNodes to completedNodes?  Not sure if having the header show NSArray as the return type will work seamlessly if the implementation actually returns NSMutableArray.